net/retriever/retriever.go

```diff
@@ -360,6 +361,10 @@ func (m *MessageRetriever) handleMessage(env pb.Envelope, addr string, id *peer.
 		} else {
 			log.Errorf("Error serializing offline message %s for storage")
 		}
+	} else if env.Message.MessageType == pb.Message_ORDER && err != nil && resp != nil {
```

Comment:
The net retriever probably shouldn't be aware of the structure of your messages. To hide this from the retriever, you can:
`env.IsOrderMessage()` 
If this is the sort of thing that's happening often and we're checking types of messages from outside the envelope, then the envelope should have `env.GetMessageType()` which returns a type which matches whatever `pb.Message_ORDER` is and then do your `==` check.